### PR TITLE
Fix Windows customization on vmware-template/create

### PR DESCRIPTION
**Pull request by sodar**
It is more of a workaround, than a fix 😞 

To do:
- [ ] change base template to `Template-WinSrv2016-20180420-0053`

### CODE REVIEW

**Comment by sodar** [comment]
Started `create-builder-template`: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/106/

Started `create-testbed-template`: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-testbed-template/31/

**Comment by sodar** [comment]
Started `create-builder-template`: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/113

File: ansible/roles/vmware-template/tasks/create.yml [hunk 1]
```diff
@@ -1,80 +1,127 @@
...
+    name: "{{ template_name }}"
+
+# NOTE: "Workaround". One way to check if customization has finished is to check hostname through
+#       VMware Tools, but vmware_guest_facts does not support this.
+#       Need to research it.
+- name: Wait for customization to finish (workaround by sleeping)
+  wait_for:
+    timeout: 900
+
+- name: Wait for correct IP address
```
Comment (PawelMarchewkaCodiLime) [inline ansible/roles/vmware-template/tasks/create.yml]: Can we wait for correct IP address after reboot? IP address may change after reboot.

File: ansible/roles/vmware-template/tasks/create.yml [hunk 2]
```diff
@@ -1,80 +1,127 @@
...
+- name: Wait for VMware tools to come up after cloning base template
+  vmware_guest_tools_wait:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: "{{ vcenter_validate_certs }}"
+    folder: "{{ template_folder }}"
+    name: "{{ template_name }}"
+
+# NOTE: "Workaround". One way to check if customization has finished is to check hostname through
```
Comment (PawelMarchewkaCodiLime) [inline ansible/roles/vmware-template/tasks/create.yml]: What about customization during the deployment of template? Shouldn't we also wait for customization?

File: ansible/vmware-create-template.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 
 - hosts: localhost
+  connection: local
```
Comment (PawelMarchewkaCodiLime) [inline ansible/vmware-create-template.yml]: Why is this 'connection: local' and adding localhost to inventory required?

File: ansible/roles/vmware-template/tasks/create.yml (hunk 2 above)
Comment (sodar) [reply ansible/roles/vmware-template/tasks/create.yml]: Playbook responsible for template deployment does not do any reboot shenanigans, which are done in create template playbook. After deploying a VM from template it waits for IP of the machine, so it should work just fine.

**Comment by sodar** [comment]
Started `create-builder-template`: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/114/

**Comment by krdln** [comment]
What's the reason for removal of `add_disk_to_vm.py` in https://github.com/Juniper/contrail-windows-ci/pull/131/commits/e1e3093503ab53d956e708030d60ff232669395b? Was it dead code?

**Comment by sodar** [comment]
@krdln Its purpose could be accomplished by simply adding another operationg with `vmware_guest` https://github.com/Juniper/contrail-windows-ci/pull/131/commits/e1e3093503ab53d956e708030d60ff232669395b#diff-8db0482cf6c387c99f0cbc0144880be4R73

**Comment by sodar** [comment]
recheck no bug

File: ansible/roles/vmware-template/tasks/create.yml [hunk 4]
```diff
@@ -90,15 +70,19 @@
...
-  with_items: "{{ vm_hdd }}"
+  vmware_guest:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: "{{ vcenter_validate_certs }}"
+    datacenter: "{{ vcenter_datacenter }}"
+    folder: "{{ template_folder }}"
+    name: "{{ template_name }}"
+    disk:
```
Comment (PawelMarchewkaCodiLime) [inline ansible/roles/vmware-template/tasks/create.yml]: Please try to get list of disks from vmware_guest_facts and append new additional disks to it.

**Pull request merged**